### PR TITLE
Feature: add Constant Contact add-on to the form migration process

### PR DESCRIPTION
**Pull request by JoshuaHungDinh**
Resolves: [GIVE-176]
## Description

This PR includes a form migration step for the new Constant Contact block migrating from a v2 form to a v3 form. I also removed the commented section of supported addons for Constant Contact so that it will display on our list when upgrading.

 Cases when the block should be added.
- While  Constant Contact  is globally enabled on all forms, if the form is not explicitly disabled from the per form setting migrating should add the block.
- While  Constant Contact  is not globally enabled if it is explicitly enabled on the per form setting than migrating should add the block.

## Testing Instructions with updated visuals

https://github.com/impress-org/givewp/assets/75056371/f6ba3298-8a3f-409d-943d-53f4f030d586

https://github.com/impress-org/givewp/assets/75056371/c4c31306-fd11-4e6b-8a94-0e41e4b74854

- Use [New  Constant Contact  Block 
](https://github.com/impress-org/give-constant-contact/pull/42)
- Set your global settings
- add a v2 form -> then migrate the form without editing the form settings.
- Verify the v3 form has the correct settings set from the global options.
- Create another v2 form -> update the per-form settings on the right -> then migrate the form
- Verify the v3 form has the correct settings set from the form options.

## Pre-review ChecklistT

-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [x] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

[GIVE-136]: https://stellarwp.atlassian.net/browse/GIVE-136?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[GIVE-64]: https://stellarwp.atlassian.net/browse/GIVE-64?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[GIVE-176]: https://stellarwp.atlassian.net/browse/GIVE-176?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/FormMigration/Steps/ConstantContact.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+/**
+ * @unreleased
+ */
+class ConstantContact extends FormMigrationStep
+{
+    /**
+     * @unreleased
+     */
+    public function canHandle(): bool
```
Comment (jonwaldstein) [inline src/FormMigration/Steps/ConstantContact.php:16]: Let's create unit tests for these

**Review by jonwaldstein — CHANGES_REQUESTED** [review]

File: src/FormMigration/Steps/ConstantContact.php (hunk 1 above)
Comment (JoshuaHungDinh) [reply src/FormMigration/Steps/ConstantContact.php:16]: Hey Jon, I've added the tests for this. I've also moved the global options into the default prop for the getMeta method. Resolved: 16ab4da1e2f98a94a99464019f9847ba6baac6fc & df77f5c8d89b352ae99644a890efa07f3e6b6c1f

Additionally while rearranging the CC methods to all be together some of the tags got mixed between them & Mailchimp. Ive fixed them in this commit : 607a2bac0520fcd2c82442b14ff6b88131f4c41f

**Review by jonwaldstein — APPROVED** [review]
@JoshuaHungDinh great job man!! You are a unit test writing machine now 🤩

**Review by rickalday — APPROVED** [review]
Passed QA tests.

**Pull request merged**